ui/app/routes/vault/cluster/secrets/backend/secret-edit.js

```diff
@@ -251,7 +251,20 @@ export default Route.extend(UnloadModelRoute, {
     if (modelType === 'secret-v2') {
       // after the the base model fetch, kv-v2 has a second associated
       // version model that contains the secret data
-      secretModel = await this.fetchV2Models(capabilities, secretModel, params);
+
+      // if no read access to metadata, return current Version from secret data endpoint's customMetadata.
+      if (!secretModel.currentVersion) {
+        let adapter = this.store.adapterFor('secret-v2-version');
+        try {
+          secretModel.currentVersion = await adapter.getSecretDataVersion(backend, secret);
```

Comment:
Need to assign currentVersion before performing fetchV2Models as it uses the currentVersion in the method.

Reply:
This fixes the error we were seeing when no read access to metadata and it was not fetching the most recent version of data or transition.